### ISSUE TITLE
Feature request: Show size of future on hover

### ISSUE THREAD

**Issue by jplatte** [rust-lang/rust-analyzer#15220]
It would be very useful if the "show type size on hover" functionality was available to anonymous futures, i.e. the types of `async` blocks and return types of `async fn`s (when hovering the function name, I guess).

**Comment by HKalbasi** [comment]
For supporting this we need to desugar async blocks into generators, which is probably also useful for some other features as well (e.g. mir).

**Comment by d3d9** [comment]
I'd love this since it would help so much with the manual optimization of futures which is still often required especially on embedded.

Since three years have passed, I was wondering if there have been changes in rust-analyzer that would make the implementation easier nowadays?

**Comment by ChayimFriedman2** [comment]
> Since three years have passed, I was wondering if there have been changes in rust-analyzer that would make the implementation easier nowadays?

No. Not only that, this has no possible path to implementation (although such path might be available in the future). For the information to be useful, it needs to be derived from the same code as rustc. Normal layout has that, async layout requires sharing MIR and async desugaring with rustc, which does not depend on r-a alone but also on T-compiler.
